### PR TITLE
Add Hello Header example

### PR DESCRIPTION
**Pull request by eguven**
Example setting custom headers.

Fixes https://github.com/gotham-rs/gotham/issues/117. I'm very new to Rust so please do point out if there are more idiomatic ways of doing this or if I need to extend the example further.

### CODE REVIEW

**Comment by bradleybeddoes** [comment]
G'day @eguven welcome to the Gotham community (and Rust!), thanks for your contribution here 👍.

I'm a day or so behind reviews but just a quick note here that it would be great if your example could use a `router` function similar to [this one](https://github.com/gotham-rs/gotham/blob/master/examples/hello_router/src/main.rs#L26). 

I believe most folks will be looking to work with Gotham via the `Router` so having this appear in all of our examples (other than `hello_world` naturally!!) seems like a good pattern to carry throughout the crates we're shipping in the `examples` directory.

**Comment by eguven** [comment]
I added the `Router` change and moved the example to after `hello_router` in readme.

File: examples/hello_header/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+[package]
+name = "hello_header"
+version = "0.1.0"
+authors = ["Bradley Beddoes <bradleybeddoes@gmail.com>",
+           "Shaun Mangelsdorf <s.mangelsdorf@gmail.com>"]
```
Comment (bradleybeddoes) [inline examples/hello_header/Cargo.toml]: Don't forget to add your name here!

**Comment by bradleybeddoes** [comment]
Hi @eguven just wondering if this was passed through rustfmt? (0.3.4-nightly or later).

We're going to look at making this a requirement of the Gotham repository CI process before 0.2 for all submissions so it would be great to start now!.

I know you said you're new to Rust so if you haven't come across rustfmt yet here are the details https://github.com/rust-lang-nursery/rustfmt.

**Comment by eguven** [comment]
Yea, it turns out I had the deprecated rustfmt 0.10.0.

    $ rustup run nightly rustfmt -V
    0.3.6-nightly (e0e3e22 2018-01-18)

File: examples/hello_header/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+// this macro defines our custom header
+header! { (GothamHeader, "X-Gotham") => [String] }
+
+/// Create a `Handler` that adds a custom header.
+pub fn say_hello_through_header(state: State) -> (State, Response) {
+    let mut res = create_response(
+        &state,
+        StatusCode::Ok,
+        Some((b"".to_vec(), mime::TEXT_PLAIN)),
```
Comment (bradleybeddoes) [inline examples/hello_header/src/main.rs]: oops I missed this earlier, all this needs to be is a `None` as you don't want/need a body + mime type in this response.

**Pull request merged**